opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -2051,20 +2051,51 @@ void CpGrid::addLgrsUpdateLeafView(const std::vector<std::array<int,3>>& cells_p
             active_cells_to_be_refined.push_back(activeCells[elemIdx]);
             this-> mark(1, cpgrid::Entity<0>(*data_[0], activeCells[elemIdx], true));
         }
-    }
+        }*/
 
-    // Check all the cells to be refined have no NNC (no neighbouring connections).
-    if ((*data_[0]).hasNNCs(active_cells_to_be_refined)){
-        OPM_THROW(std::logic_error, "NNC face on a cell containing LGR is not supported yet.");
+    
+
+    // Determine the assigned level for the refinement of each marked cell
+    std::vector<int> assignRefinedLevel(data_[0]->size(0));
+    for(const auto& element: elements(this->leafGridView())) {
+        std::cout<< "elemIdx: " << element.index() <<  " global: " << globalCell()[element.index()] << std::endl;
+        std::array<int,3> ijk;
+        getIJK(element.index(), ijk);
+        for (int level = 0; level < static_cast<int>(startIJK_vec.size()); ++level) {
+            bool belongsToLevel = true;
+            for (int c = 0; c < 3; ++c) {
+                belongsToLevel = belongsToLevel && ( (ijk[c] >= startIJK_vec[level][c]) && (ijk[c] <= endIJK_vec[level][c]) );
```

Comment:
Please double check the comparison. Shouldn't the first be `>` as startIJK is the index of the first cell to be refined? And maybe the last one  `<`?

Reply:
Great comment! That was the reason of failing tests

Reply:
Well, I am definitely wrong about startIJK. :see_no_evil:

Reply:
The comment was pointing correctly to the issue :) Nevertheless, only endIJK comparison was wrong